hw/ip/otbn/dv/otbnsim/sim/sim.py

```diff
@@ -27,10 +30,18 @@ def run(self, verbose: bool) -> int:
 
         '''
         insn_count = 0
+        # ISS will stall at start until URND data is valid, immediately set it
+        # valid when in free running mode as nothing else will.
+        self.state.set_urnd_reseed_complete()
```

Comment:
Hmm, I wonder whether we should hoist the definition of `run` out of `sim.py` and into `standalone.py`. That way, the code in `sim.py` doesn't need to have an opinion of whether it's interacting with other components or not.

Reply:
On the other hand it's nice if `OTBNSim` can support a 'just run my code' function (i.e. `run`) without needing extra support. Currently only `standalone.py` needs that but some future thing using `OTBNSim` as a straight ISS that doesn't care about RND interactions that isn't `standalone.py` is plausible.

Reply:
Yeah, you're right. I think my preference for doing that would be a "StandaloneSim" class (used by `standalone.py` and whatever else we want it for). I'd still rather not put that logic in `sim.py` long-term.

Reply:
Please could you file a follow-up issue to remind us (me!) to do this at some point?